.github/workflows/docker-push.yml

```diff
@@ -3,35 +3,44 @@ on: create
 
 jobs:
   release:
+    if: github.event.ref_type == 'tag'
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2.4.0
-        if: github.event.ref_type == 'tag'
 
       - uses: actions/setup-go@v2.1.5
-        if: github.event.ref_type == 'tag'
+
         id: go
         with:
           go-version: 1.17
 
       - name: go-build
-        if: github.event.ref_type == 'tag'
+
         run: go build -o fake-gcs-server
         env:
           CGO_ENABLED: 0
 
       - name: sanity-check
-        if: github.event.ref_type == 'tag'
         uses: docker://alpine
         with:
           entrypoint: sh
           args: -c "./fake-gcs-server -h"
 
-      - name: docker-publish
-        if: github.event.ref_type == 'tag'
-        uses: fsouza/docker-publish-action@v1.0.1
+      - name: setup qemu
+        uses: docker/setup-qemu-action@v1
+      - name: setup buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+      - name: available platforms
+        run: echo ${{ steps.buildx.outputs.platforms }}
+      - name: login to docker hub
+        uses: docker/login-action@v1
         with:
-          dockerfile: ci/Dockerfile
-          docker-image: fsouza/fake-gcs-server
-          DOCKER-USERNAME: ${{ secrets.DOCKER_USERNAME }}
-          docker-password: ${{ secrets.DOCKER_PASSWORD }}
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+      - name: build the image
+        run: |
+          docker buildx build \
+          --push \
+          --tag ${{ secrets.DOCKER_USERNAME }}/fake-gcs-server:latest \
```

Comment:
Nice, this looks good. I guess one thing missing would be the tagging scheme (for example, when we tag v1.2.3, we push `fsouza/fake-gcs-server:1`, `fsouza/fake-gcs-server:1.2` and `fsouza/fake-gcs-server:1.2.3`). 

Looks like docker provides another GH action that can accomplish this: https://github.com/docker/metadata-action#semver

I can take of it though, if you prefer. Up to you.

Reply:
Oh, yeah. totally forgot this. I'll have a look later tonight 👍